Project structure improvements

magyarosibotond opened the pull request:
I've made some project structure improvements for a cleaner overview.
- **cocoapods** in added to a Gemfile to enforce the use of CocoaPods via bundler. Version management is easier this way when working in a team.
- Replace folder references with folder groups. The Xcode project structure is counterpart of the file system structure so it's easier to find files when using a browser. It also makes the _MonitorizareVot_ folder cleaner.
- Removed some old and unused CocoaPods artifacts.

Commit:
Install Cocaopods through bundler

Commit:
Remove old, unused Cocoapods artifacts

Commit:
Remove SnapKit as it's not used

Commit:
Replace folder references with folder groups to have the same file structure in both xcode and the filesystem

Commit:
Remove some more old CocoaPods artifacts

Commit:
Enforce bundler when running pod

CristiHabliuc commented:
Hey thanks for taking the time to do all this, we really appreciate it!

Until I get a deeper look into the actual changes, I wish you'd have done them in separate PRs, because some things we won't be changing. From first looks, these would still stay:
- Remove SnapKit - we'll keep it since we'll use it going forward when manually building constraints
- Bundler w/ Cocoapods - I'm not sure about the necessity of this change, what was the reason behind it?

I'll checkout the PR tomorrow during the HackDay and get back with details. Thank you for the time put into this, some of these changes are really good!

magyarosibotond commented:
Hey @CristiHabliuc!

Thanks for taking the time to go through the Pull request! 

The overall purpose of the merge request was to add the same structure to the project on the filesystem level (with improvements here and there) as it is structured in the Xcode project. While doing that I've also removed some unused files and dependencies.

To answer your questions:

- I've removed `SnapKit` since it wasn't used in the app and I didn't know you're planning to use it. I'll re-add that dependency later on.
- I usually use `bundler` to manage the rubygems used by the build environment (simiar to what pyenv is to python). By using Bundler you no longer rely on the local `cocaopods` (or `fastlane`) version since it's tracked in the version control system. Relying on the local `cocoapods` (gem install) is extremely painful when team members have different versions and you need a `pod install` every time you pull. I also had a different version when I pulled the project first.

Since many files were moved it might be easier to check out the branch locally and review the changes there.
